packages/browser-service/src/browser.service.ts

```diff
@@ -0,0 +1,165 @@
+import {
+  CommonWalletConfig,
+  NetworkConfig,
+  WalletConfig,
+  WalletPage,
+  WalletTypes,
+} from '@lidofinance/wallets-testing-wallets';
+import {
+  Extension,
+  ExtensionService,
+} from '@lidofinance/wallets-testing-extensions';
+import { EthereumNodeService } from '@lidofinance/wallets-testing-nodes';
+import {
+  DEFAULT_BROWSER_CONTEXT_DIR_NAME,
+  WALLET_PAGES,
+} from './browser.constants';
+import {
+  BrowserContextService,
+  BrowserOptions,
+} from './browser.context.service';
+import { mnemonicToAccount } from 'viem/accounts';
+import { test } from '@playwright/test';
+
+type NodeConfig = {
+  rpcUrlToMock: string; // example: '**/api/rpc?chainId=1'
+};
+
+type BrowserServiceOptions = {
+  networkConfig: NetworkConfig;
+  walletConfig: WalletConfig;
+  nodeConfig: NodeConfig;
+  browserOptions?: BrowserOptions;
+  enableBrowserContext?: boolean;
+};
+
+export class BrowserService {
+  private walletPage: WalletPage<WalletTypes.EOA>;
+  private browserContextService: BrowserContextService;
+  public ethereumNodeService: EthereumNodeService;
+
+  public isFork: boolean;
+
+  constructor(private options: BrowserServiceOptions) {}
+
+  getWalletPage() {
+    return this.walletPage;
+  }
+
+  getBrowserContext() {
+    return this.browserContextService.browserContext;
+  }
+
+  getBrowserContextPage() {
+    return this.browserContextService.browserContext.pages()[0];
+  }
+
+  async initWalletSetup(useFork?: boolean) {
+    if (useFork) {
+      await this.setupWithNode();
+    } else {
+      await this.setup();
+      await this.walletPage.setupNetwork(this.options.networkConfig);
+      await this.walletPage.changeNetwork(this.options.networkConfig.chainName);
+      await this.browserContextService.closePages();
+    }
+    this.isFork = !!this.ethereumNodeService;
+  }
+
+  async setupWithNode() {
+    this.ethereumNodeService = new EthereumNodeService({
+      chainId: this.options.networkConfig.chainId,
+      rpcUrl: this.options.networkConfig.rpcUrl,
+      defaultBalance: 100,
+    });
+    await this.ethereumNodeService.startNode();
+    const account = this.ethereumNodeService.getAccount();
+    await this.setup();
+    if (!(await this.walletPage.isWalletAddressExist(account.address))) {
+      await this.walletPage.importKey(account.secretKey);
+    } else {
+      await this.walletPage.changeWalletAccountByName(account.address);
+    }
+
+    await this.walletPage.setupNetwork({
+      ...this.options.networkConfig,
+      chainName: this.options.networkConfig.chainName,
+      rpcUrl: this.ethereumNodeService.state.nodeUrl,
+    });
+
+    await this.ethereumNodeService.mockRoute(
+      this.options.nodeConfig.rpcUrlToMock,
+      this.browserContextService.browserContext,
+    );
+    await this.browserContextService.closePages();
+  }
+
+  async setup(commonWalletConfig?: CommonWalletConfig) {
```

Comment:
When we should pass `commonWalletConfig` if we already init it in `walletConfig.COMMON.commonWalletConfig`?

Reply:
How do you fill can we shortify current function? since for now it's looks like longrid, but looks like it can be logically structed